rust/template/d3log/src/broadcast.rs

```diff
@@ -0,0 +1,74 @@
+// a replication component to support broadcast metadata facts. this includes 'split horizon' as a temporary
+// fix for simple pairwise loops. This will need an additional distributed coordination mechanism in
+// order to maintain a consistent spanning tree (and a strategy for avoiding storms for temporariliy
+// inconsistent topologies
+
+use crate::{Batch, Port, Transport};
+
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::{Arc, Mutex};
+
+pub struct Ingress {
```

Comment:
If I understand correctly, this is needed so that broadcast group members can send messages to all group members excluding self.  Please document this here (and also in `trait PubSub`) for ease of understanding.